src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt

```diff
@@ -211,5 +244,32 @@ class Cargo(private val cargoExecutable: Path) {
             environmentVariables.configureCommandLine(cmdLine, true)
             return cmdLine
         }
+
+        fun checkNeedInstallRustfilt(project: Project): Boolean =
+            checkNeedInstallPackage(project, "rustfilt")
+
+        private fun checkNeedInstallPackage(project: Project, packageName: String): Boolean {
+            fun isNotInstalled(): Boolean {
+                val cargo = project.toolchain?.rawCargo() ?: return false
+                val installed = cargo.listPackages().any { it.startsWith(packageName) }
+                return !installed
+            }
+
+            val needInstall = if (ApplicationManager.getApplication().isDispatchThread) {
+                project.computeWithCancelableProgress("Checking if $packageName is installed...", ::isNotInstalled)
+            } else {
+                isNotInstalled()
+            }
+
+            if (needInstall) {
+                project.showBalloon(
+                    "$packageName is not installed",
+                    NotificationType.ERROR,
```

Comment:
But it's not an error in case of profiler because we can successfully invoke all profiler-related action. I think it's should be warning. Also, it would be great to provide more detailed message why we need this tool at all.
So let's extend API of this method to pass custom `NotificationType` and additional message info if needed